feat: ROOT-16: Support reading Parquet from source cloud storages in LSE

matt-bernstein opened the pull request:
fix test mock
<!--

This description MUST be filled out for a PR to receive a review. Its primary purposes are:

 - to enable your reviewer to review your code easily, and
 - to convince your reviewer that your code works as intended.

Some pointers to think about when filling out your PR description:
 - Reason for change: Description of problem and solution
 - Screenshots: All visible changes should include screenshots.
 - Rollout strategy: How will this code be rolled out? Feature flags / env var / other
 - Testing: Description of how this is being verified
 - Risks: Are there any known risks associated with this change, eg to security or performance?
 - Reviewer notes: Any info to help reviewers approve the PR
 - General notes: Any info to help onlookers understand the code, or callouts to significant portions.

You may use AI tools such as Copilot Actions to assist with writing your PR description (see https://docs.github.com/en/copilot/using-github-copilot/using-github-copilot-for-pull-requests/creating-a-pull-request-summary-with-github-copilot); however, an AI summary isn't enough by itself. You'll need to provide your reviewer with strong evidence that your code works as intended, which requires actually running the code and showing that it works.

-->

matt-bernstein committed:
feat: ROOT-16: Support reading Parquet files in source cloud storages in LSE

matt-bernstein committed:
gcs mock bugfix

jombooth commented on label_studio/tests/utils.py:
```diff
@@ -124,7 +124,7 @@ def generate_signed_url(self, **kwargs):
             if self.is_json:
-                return json.dumps(self.sample_json_contents)
+                return json.dumps(self.sample_json_contents).encode('utf-8')
             return data
```
should `data` also be returned encoded?

matt-bernstein replied:
that codepath is never hit currently, but for consistency, yes

jombooth reviewed (approved):
just one question

matt-bernstein committed:
make method consistent

jombooth commented on label_studio/tests/utils.py:
```diff
@@ -124,8 +124,8 @@ def generate_signed_url(self, **kwargs):
         def download_as_bytes(self):
```
slightly DRY-er approach

```
def download_as_bytes(self):
    data = json.dumps(self.sample_json_contents) if self.is_json else f'test_blob_{self.key}'
    return data.encode()
```

matt-bernstein replied:
would agree in isolation, but want to make the parallel with `download_as_string` very obvious

jombooth replied:
ah, even simpler then would be just:

```
def download_as_bytes(self):
    return self.download_as_string().encode()
```

matt-bernstein replied:
fair enough :)

matt-bernstein committed:
nicer impl

matt-bernstein committed:
Merge remote-tracking branch 'origin/develop' into fb-ROOT-16

matt-bernstein commented:
/fm sync
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/15216702267)

matt-bernstein committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/15216702267

matt-bernstein committed:
rename param

jombooth commented:
/git merge develop
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/15233912348)
> Successfully merged:  create mode 100644 docs/source/guide/labeling_bulk.md

jombooth committed:
Merge branch 'develop' into 'fb-ROOT-16'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/15233912348
